### DISCUSSION TITLE
Issues with HEIC Support on AIO 7.0

### DISCUSSION THREAD

**Discussion by Byter3** [nextcloud/all-in-one#3209] [Questions] [answered]
I know I'm gonna be very unpopular... :(

Issues with heic thumbnails after upgrading to AIO v7.0
Got two thing in the log which may be the cause:

`
[index] Error: OCP\Files\NotPermittedException: Could not create folder "/appdata_octnlljryifn/preview/a/7/3/f/b/0/8/1842217" at <<closure>>

 0. /var/www/html/lib/private/Files/AppData/AppData.php line 147
    OC\Files\Node\Folder->newFolder("a/7/3/f/b/0/8/1842217")
 1. /var/www/html/lib/private/Preview/Storage/Root.php line 74
    OC\Files\AppData\AppData->newFolder("a/7/3/f/b/0/8/1842217")
 2. /var/www/html/lib/private/Preview/Generator.php line 643
    OC\Preview\Storage\Root->newFolder("1842217")
 3. /var/www/html/lib/private/Preview/Generator.php line 139
    OC\Preview\Generator->getPreviewFolder(["OC\\Files\\Node\\File"])
 4. /var/www/html/lib/private/Preview/Generator.php line 116
    OC\Preview\Generator->generatePreviews(["OC\\Files\\Node\\File"], [[64,64,false,"fill"]], "video/mp4")
 5. /var/www/html/lib/private/PreviewManager.php line 192
    OC\Preview\Generator->getPreview(["OC\\Files\\Node\\File"], 64, 64, false, "fill", null)
 6. /var/www/html/apps/photos/lib/Controller/PreviewController.php line 162
    OC\PreviewManager->getPreview(["OC\\Files\\Node\\File"], 64, 64)
 7. /var/www/html/apps/photos/lib/Controller/PreviewController.php line 128
    OCA\Photos\Controller\PreviewController->fetchPreview(["OC\\Files\\Node\\File"], 64, 64)
 8. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
    OCA\Photos\Controller\PreviewController->index(1842217, 64, 64)
 9. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
    OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Photos\\C ... "], "index")
10. /var/www/html/lib/private/AppFramework/App.php line 183
    OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Photos\\C ... "], "index")
11. /var/www/html/lib/private/Route/Router.php line 315
    OC\AppFramework\App::main("OCA\\Photos\\Co ... r", "index", ["OC\\AppFramewo ... "], ["1842217","photos.preview.index"])
12. /var/www/html/lib/base.php line 1071
    OC\Route\Router->match("/apps/photos/api/v1/preview/1842217")
13. /var/www/html/index.php line 36
    OC::handleRequest()

GET /apps/photos/api/v1/preview/1842217?etag=64dcf7acb406a&x=64&y=64
`

`
[no app in context] Error: GuzzleHttp\Exception\ClientException: Client error: `POST http://nextcloud-aio-imaginary:9000/pipeline?operations=%5B%7B%22operation%22%3A%22autorotate%22%7D%2C%7B%22operation%22%3A%22fit%22%2C%22params%22%3A%7B%22width%22%3A2048%2C%22height%22%3A2048%2C%22stripmeta%22%3A%22true%22%2C%22type%22%3A%22jpeg%22%2C%22norotation%22%3A%22true%22%2C%22quality%22%3A%2260%22%7D%7D%5D` resulted in a `406 Not Acceptable` response:
{"message":"Unsupported media type","status":406}
 at <<closure>>

 0. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/guzzle/src/Middleware.php line 72
    GuzzleHttp\Exception\RequestException::create("*** sensitive parameters replaced ***")
 1. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 209
    GuzzleHttp\Middleware::GuzzleHttp\{closure}("*** sensitive parameters replaced ***")
 2. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 158
    GuzzleHttp\Promise\Promise::callHandler(1, "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***")
 3. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/TaskQueue.php line 52
    GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}("*** sensitive parameters replaced ***")
 4. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 251
    GuzzleHttp\Promise\TaskQueue->run(true)
 5. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 227
    GuzzleHttp\Promise\Promise->invokeWaitFn()
 6. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 272
    GuzzleHttp\Promise\Promise->waitIfPending()
 7. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 229
    GuzzleHttp\Promise\Promise->invokeWaitList()
 8. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 69
    GuzzleHttp\Promise\Promise->waitIfPending()
 9. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/guzzle/src/Client.php line 189
    GuzzleHttp\Promise\Promise->wait()
10. /var/www/html/lib/private/Http/Client/Client.php line 297
    GuzzleHttp\Client->request("post", "http://nextclou ... e", ["/mnt/ncdata/fi ... "])
11. /var/www/html/lib/private/Preview/Imaginary.php line 147
    OC\Http\Client\Client->post("http://nextclou ... e", [["[{\"operation ... "])
12. /var/www/html/lib/private/Preview/GeneratorHelper.php line 62
    OC\Preview\Imaginary->getCroppedThumbnail(["OC\\Files\\Node\\File"], 2048, 2048, false)
13. /var/www/html/lib/private/Preview/Generator.php line 397
    OC\Preview\GeneratorHelper->getThumbnail(["OC\\Preview\\Imaginary"], ["OC\\Files\\Node\\File"], 2048, 2048)
14. /var/www/html/lib/private/Preview/Generator.php line 373
    OC\Preview\Generator->generateProviderPreview(["OC\\Files\\SimpleFS\\SimpleFolder"], ["OC\\Files\\Node\\File"], 2048, 2048, false, true, "image/jpeg", "")
15. /var/www/html/lib/private/Preview/Generator.php line 166
    OC\Preview\Generator->getMaxPreview(["OC\\Files\\SimpleFS\\SimpleFolder"], "*** sensitive parameters replaced ***", ["OC\\Files\\Node\\File"], "image/jpeg", "")
16. /var/www/html/lib/private/Preview/Generator.php line 116
    OC\Preview\Generator->generatePreviews(["OC\\Files\\Node\\File"], [[512,512,false,"fill"]], "image/jpeg")
17. /var/www/html/lib/private/PreviewManager.php line 192
    OC\Preview\Generator->getPreview(["OC\\Files\\Node\\File"], 512, 512, false, "fill", "*** sensitive parameters replaced ***")
18. /var/www/html/apps/photos/lib/Controller/PreviewController.php line 162
    OC\PreviewManager->getPreview(["OC\\Files\\Node\\File"], 512, 512)
19. /var/www/html/apps/photos/lib/Controller/PreviewController.php line 128
    OCA\Photos\Controller\PreviewController->fetchPreview(["OC\\Files\\Node\\File"], 512, 512)
20. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
    OCA\Photos\Controller\PreviewController->index(1842262, 512, 512)
21. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
    OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Photos\\C ... "], "index")
22. /var/www/html/lib/private/AppFramework/App.php line 183
    OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Photos\\C ... "], "index")
23. /var/www/html/lib/private/Route/Router.php line 315
    OC\AppFramework\App::main("OCA\\Photos\\Co ... r", "index", ["OC\\AppFramewo ... "], ["1842262","photos.preview.index"])
24. /var/www/html/lib/base.php line 1071
    OC\Route\Router->match("/apps/photos/api/v1/preview/1842262")
25. /var/www/html/index.php line 36
    OC::handleRequest()

GET /apps/photos/api/v1/preview/1842262?etag=64dd8086bc806&x=512&y=512
`

`
[no app in context] Error: GuzzleHttp\Exception\ClientException: Client error: `POST http://nextcloud-aio-imaginary:9000/pipeline?operations=%5B%7B%22operation%22%3A%22autorotate%22%7D%2C%7B%22operation%22%3A%22fit%22%2C%22params%22%3A%7B%22width%22%3A256%2C%22height%22%3A256%2C%22stripmeta%22%3A%22true%22%2C%22type%22%3A%22jpeg%22%2C%22norotation%22%3A%22true%22%2C%22quality%22%3A%2260%22%7D%7D%5D` resulted in a `406 Not Acceptable` response:
{"message":"Unsupported media type","status":406}
 at <<closure>>

 0. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/guzzle/src/Middleware.php line 72
    GuzzleHttp\Exception\RequestException::create("*** sensitive parameters replaced ***")
 1. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 209
    GuzzleHttp\Middleware::GuzzleHttp\{closure}("*** sensitive parameters replaced ***")
 2. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 158
    GuzzleHttp\Promise\Promise::callHandler(1, "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***")
 3. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/TaskQueue.php line 52
    GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}("*** sensitive parameters replaced ***")
 4. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 251
    GuzzleHttp\Promise\TaskQueue->run(true)
 5. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 227
    GuzzleHttp\Promise\Promise->invokeWaitFn()
 6. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 272
    GuzzleHttp\Promise\Promise->waitIfPending()
 7. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 229
    GuzzleHttp\Promise\Promise->invokeWaitList()
 8. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 69
    GuzzleHttp\Promise\Promise->waitIfPending()
 9. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/guzzle/src/Client.php line 189
    GuzzleHttp\Promise\Promise->wait()
10. /var/www/html/lib/private/Http/Client/Client.php line 297
    GuzzleHttp\Client->request("post", "http://nextclou ... e", ["/mnt/ncdata/fi ... "])
11. /var/www/html/lib/private/Preview/Imaginary.php line 147
    OC\Http\Client\Client->post("http://nextclou ... e", [["[{\"operation ... "])
12. /var/www/html/lib/private/Preview/GeneratorHelper.php line 62
    OC\Preview\Imaginary->getCroppedThumbnail(["OC\\Files\\Node\\File"], 256, 256, false)
13. /var/www/html/lib/private/Preview/Generator.php line 397
    OC\Preview\GeneratorHelper->getThumbnail(["OC\\Preview\\Imaginary"], ["OC\\Files\\Node\\File"], 256, 256)
14. /var/www/html/lib/private/Preview/Generator.php line 255
    OC\Preview\Generator->generateProviderPreview(["OC\\Files\\SimpleFS\\SimpleFolder"], ["OC\\Files\\Node\\File"], 256, 256, false, false, "image/jpeg", "")
15. /var/www/html/lib/private/Preview/Generator.php line 155
    OC\Preview\Generator->getSmallImagePreview(["OC\\Files\\SimpleFS\\SimpleFolder"], "*** sensitive parameters replaced ***", ["OC\\Files\\Node\\File"], "image/jpeg", "", false)
16. /var/www/html/lib/private/Preview/Generator.php line 116
    OC\Preview\Generator->generatePreviews(["OC\\Files\\Node\\File"], [[64,64,false,"fill"]], "image/jpeg")
17. /var/www/html/lib/private/PreviewManager.php line 192
    OC\Preview\Generator->getPreview(["OC\\Files\\Node\\File"], 64, 64, false, "fill", "*** sensitive parameters replaced ***")
18. /var/www/html/apps/photos/lib/Controller/PreviewController.php line 162
    OC\PreviewManager->getPreview(["OC\\Files\\Node\\File"], 64, 64)
19. /var/www/html/apps/photos/lib/Controller/PreviewController.php line 128
    OCA\Photos\Controller\PreviewController->fetchPreview(["OC\\Files\\Node\\File"], 64, 64)
20. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
    OCA\Photos\Controller\PreviewController->index(1842262, 64, 64)
21. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
    OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Photos\\C ... "], "index")
22. /var/www/html/lib/private/AppFramework/App.php line 183
    OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Photos\\C ... "], "index")
23. /var/www/html/lib/private/Route/Router.php line 315
    OC\AppFramework\App::main("OCA\\Photos\\Co ... r", "index", ["OC\\AppFramewo ... "], ["1842262","photos.preview.index"])
24. /var/www/html/lib/base.php line 1071
    OC\Route\Router->match("/apps/photos/api/v1/preview/1842262")
25. /var/www/html/index.php line 36
    OC::handleRequest()

GET /apps/photos/api/v1/preview/1842262?etag=64dd8086bc806&x=64&y=64
`

**Comment by szaimen** [comment]
> Error: OCP\Files\NotPermittedException: Could not create folder "/appdata_octnlljryifn/preview/a/7/3/f/b/0/8/1842217

This one looks suspicious. Can you verify that permissions are set correctly for the appdata directory?

**Comment by Byter3** [comment]
Please excuse my ignorance, but I'm using AIO as is. It should be the default path, with default privileges.

**Comment by szaimen** [comment]
> Please excuse my ignorance, but I'm using AIO as is. It should be the default path, with default privileges.

Can you run the following command: `sudo docker exec -it nextcloud-aio-nextcloud chown -R www-data:www-data /mnt/ncdata/appdata*` and check if `Error: OCP\Files\NotPermittedException: Could not create folder "/appdata_octnlljryifn/preview/a/7/3/f/b/0/8/1842217` comes up again?

**Comment by Byter3** [comment]
Thank you!
Ran it, I will monitor the logs to see if it comes up again, I quickly checked in the files app and heic thumbnails are still missing which had thumbnails prior to the upgrade. Do I have to re run a the pregeneration everytime I upgrade the AIO container or what am I missing here?

**Comment by szaimen** [comment]
I just uploaded a heic image on my test instance and it shows a preview. 
![image](https://github.com/nextcloud/all-in-one/assets/42591237/5d88be01-14ad-4417-bcb2-19c4a1d9906c)


So not sure what is wrong in your case. Maybe one of your installed apps blocks the previews?

**Comment by szaimen** [comment]
For example, can you try to disable Fulltextsearch and check if it helps? (Not that I think that it should but still worth a try).

**Comment by Byter3** [comment]
> I just uploaded a heic image on my test instance and it shows a preview. So not sure what is wrong in your case. Maybe one of your installed apps blocks the previews?

Thanks for the tip, I might disable everything and give it a go. Even with the previouse AIO version HEIF was very intermitent. Sometimes it loaded sometimes it not.


> For example, can you try to disable Fulltextsearch and check if it helps? (Not that I think that it should but still worth a try).

Nope!

**Comment by Byter3** [comment]
Also maybe a hint, that I think PDFs also should have a preview thumbnail, but they don't have it also. The only place where I can see heif thumbnails is in the memories app.

**Comment by szaimen** [comment]
PDF files have indeed a preview on my test server: 
![image](https://github.com/nextcloud/all-in-one/assets/42591237/52ff90ae-5cce-4afb-8f00-fd84510afd36)

**Comment by Byter3** [comment]
Alright, than it is not working also for me.

**Comment by szaimen** [comment]
So I suspect one of your installed apps to block previews. You could try to disable them one by one and check if it makes things work. Also you might try to upload a new file and check if preview works for that?

**Comment by szaimen** [comment]
BTW, are you using the external storage app or just normal storage? Also did you maybe disable previews in your config.php?

**Comment by szaimen** [comment]
I'll move the issue to discussions since I cannot reproduce the problem.

**Comment by Byter3** [comment]
Yes, I always check if there is a difference with new files and already existing ones or the on the interanl storage or the mounted SMB.


> BTW, are you using the external storage app or just normal storage? Also did you maybe disable previews in your config.php?

I use both, constantly checking both sites.

**Reply by szaimen** [reply]
So did you maybe disable preview for that specific SMB mount? See https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/external_storage_configuration_gui.html?highlight=preview#mount-options

**Reply by Byter3** [reply]
Ofc not

**Reply by szaimen** [reply]
okay just asking

**Comment by Byter3** [comment]
> BTW, are you using the external storage app or just normal storage? Also did you maybe disable previews in your config.php?

Did not change it, but will check. Somebody already raised this in an other discussion. Here: https://github.com/nextcloud/photos/issues/1961#issuecomment-1676149173

Also a tremendous thank you for all the help. I will go and disable my installed apps one by one tomorrow and see if that helps.

**Reply by szaimen** [reply, marked as answer]
> Here: [nextcloud/photos#1961 (comment)](https://github.com/nextcloud/photos/issues/1961#issuecomment-1676149173)

I rather dont think that this solves the problem as you already installed imaginary which takes care of heic images.

> I will go and disable my installed apps one by one tomorrow and see if that helps.

sounds good

**Comment by szaimen** [comment]
Another question: is this external storage created by the admin or by the user?

**Reply by Byter3** [reply]
By the user. But as I said, I also checking it with internal files as well and they give the same result all the time. If heic thumbnails loads on the internal it will load an the SMB share as well. If not then it won't load anywhere.

**Comment by Byter3** [comment]
It's very tricky to figure it out what could be the cause of it, since sometimes the thumbnails load and sometimes they aren't.
There is one new clue in the logs however:
(Today the thumbnails where there, but after I rechecked them a couple of minutes later, they weren't there anymore. I checked the log and thius is what I found)

`
[no app in context] Error: GuzzleHttp\Exception\ClientException: Client error: `POST http://nextcloud-aio-imaginary:9000/pipeline?operations=%5B%7B%22operation%22%3A%22autorotate%22%7D%2C%7B%22operation%22%3A%22fit%22%2C%22params%22%3A%7B%22width%22%3A2048%2C%22height%22%3A2048%2C%22stripmeta%22%3A%22true%22%2C%22type%22%3A%22jpeg%22%2C%22norotation%22%3A%22true%22%2C%22quality%22%3A%2260%22%7D%7D%5D` resulted in a `406 Not Acceptable` response:
{"message":"Unsupported media type","status":406}
 at <<closure>>

 0. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/guzzle/src/Middleware.php line 72
    GuzzleHttp\Exception\RequestException::create("*** sensitive parameters replaced ***")
 1. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 209
    GuzzleHttp\Middleware::GuzzleHttp\{closure}("*** sensitive parameters replaced ***")
 2. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 158
    GuzzleHttp\Promise\Promise::callHandler(1, "*** sensitive parameters replaced ***", "*** sensitive parameters replaced ***")
 3. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/TaskQueue.php line 52
    GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}("*** sensitive parameters replaced ***")
 4. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 251
    GuzzleHttp\Promise\TaskQueue->run(true)
 5. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 227
    GuzzleHttp\Promise\Promise->invokeWaitFn()
 6. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 272
    GuzzleHttp\Promise\Promise->waitIfPending()
 7. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 229
    GuzzleHttp\Promise\Promise->invokeWaitList()
 8. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/promises/src/Promise.php line 69
    GuzzleHttp\Promise\Promise->waitIfPending()
 9. /var/www/html/custom_apps/fulltextsearch_elasticsearch/vendor/guzzlehttp/guzzle/src/Client.php line 189
    GuzzleHttp\Promise\Promise->wait()
10. /var/www/html/lib/private/Http/Client/Client.php line 297
    GuzzleHttp\Client->request("post", "http://nextclou ... e", ["/mnt/ncdata/fi ... "])
11. /var/www/html/lib/private/Preview/Imaginary.php line 147
    OC\Http\Client\Client->post("http://nextclou ... e", [["[{\"operation ... "])
12. /var/www/html/lib/private/Preview/GeneratorHelper.php line 62
    OC\Preview\Imaginary->getCroppedThumbnail(["OC\\Files\\Node\\File"], 2048, 2048, false)
13. /var/www/html/lib/private/Preview/Generator.php line 397
    OC\Preview\GeneratorHelper->getThumbnail(["OC\\Preview\\Imaginary"], ["OC\\Files\\Node\\File"], 2048, 2048)
14. /var/www/html/lib/private/Preview/Generator.php line 373
    OC\Preview\Generator->generateProviderPreview(["OC\\Files\\SimpleFS\\SimpleFolder"], ["OC\\Files\\Node\\File"], 2048, 2048, false, true, "image/jpeg", "")
15. /var/www/html/lib/private/Preview/Generator.php line 166
    OC\Preview\Generator->getMaxPreview(["OC\\Files\\SimpleFS\\SimpleFolder"], "*** sensitive parameters replaced ***", ["OC\\Files\\Node\\File"], "image/jpeg", "")
16. /var/www/html/lib/private/Preview/Generator.php line 116
    OC\Preview\Generator->generatePreviews(["OC\\Files\\Node\\File"], [[512,512,false,"fill"]], "image/jpeg")
17. /var/www/html/lib/private/PreviewManager.php line 192
    OC\Preview\Generator->getPreview(["OC\\Files\\Node\\File"], 512, 512, false, "fill", "*** sensitive parameters replaced ***")
18. /var/www/html/custom_apps/memories/lib/Controller/ImageController.php line 60
    OC\PreviewManager->getPreview(["OC\\Files\\Node\\File"], 512, 512, false, "fill")
19. /var/www/html/custom_apps/memories/lib/UtilController.php line 20
    OCA\Memories\Controller\ImageController->OCA\Memories\Controller\{closure}("*** sensitive parameters replaced ***")
20. /var/www/html/custom_apps/memories/lib/Controller/ImageController.php line 67
    OCA\Memories\Util::guardEx(["Closure"])
21. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 230
    OCA\Memories\Controller\ImageController->preview(1842262, 512, 512, true, "fill")
22. /var/www/html/lib/private/AppFramework/Http/Dispatcher.php line 137
    OC\AppFramework\Http\Dispatcher->executeController(["OCA\\Memories\ ... "], "preview")
23. /var/www/html/lib/private/AppFramework/App.php line 183
    OC\AppFramework\Http\Dispatcher->dispatch(["OCA\\Memories\ ... "], "preview")
24. /var/www/html/lib/private/Route/Router.php line 315
    OC\AppFramework\App::main("OCA\\Memories\\ ... r", "preview", ["OC\\AppFramewo ... "], ["1842262","memories.Image.preview"])
25. /var/www/html/lib/base.php line 1071
    OC\Route\Router->match("/apps/memories/ ... 2")
26. /var/www/html/index.php line 36
    OC::handleRequest()

GET /apps/memories/api/image/preview/1842262?c=64dd8086bc806&x=512&y=512&a=1
`
